hbase-server/src/main/java/org/apache/hadoop/hbase/master/HbckChore.java

```diff
@@ -223,15 +223,37 @@ private void loadRegionsFromInMemoryState() {
           .isTableState(regionInfo.getTable(), TableState.State.DISABLED)) {
         disabledTableRegions.add(regionInfo.getRegionNameAsString());
       }
-      if (regionInfo.isSplitParent()) {
+      if (regionState.isSplit()) {
```

Comment:
Most of this patch is just more logging. This is a real bug fix, though. Right now it just affects logging (numbers printed in a debug log line) but if in the future code depends on `splitParentRegions` to be properly populated, this will be needed.

Reply:
+1 on this change.

Was always odd have a Regions' split state recorded in its RegionInfo; RegionInfo does not keep internally if it is being merged or if open, etc. Regions are manipulated by other entities so state belongs elsewhere. +1 on move to regionstate checking instead.